demo/dlb-init.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+NUM_VFS_PER_PF=${NUM_VFS_PER_PF:-1}
+
+bind() {
+    BSF=$1
+    DRIVER=$2
+
+    echo "$BSF" > /sys/bus/pci/drivers/"$DRIVER"/bind;
+}
+
+unbind() {
+    BSF=$1
+    OLD_DRIVER=$2
+
+    echo -n "$BSF" > /sys/bus/pci/drivers/"$OLD_DRIVER"/unbind;
+}
+
+#get dlb pf pci ids
+DEVS=$(realpath /sys/bus/pci/drivers/dlb2/????:??:??\.0)
+skip=true
+for dev in $DEVS; do
+  SRIOV_NUMVFS_PATH=$dev/sriov_numvfs
+  #every 2nd pf is used for configuring vfs
+  if $skip; then
+    skip=false
+    echo "$dev" "skipped"
+    continue
+  else
+    skip=true
+    echo "$dev" "configured"
+    if test -w "$SRIOV_NUMVFS_PATH" -a "$(cat "$SRIOV_NUMVFS_PATH")" -eq 0 ; then 
```

Comment:
`$skip` can be tested here to simplify the flow:

```
for dev in $DEVS; do
    if test ! skip and -w and -eq 0; then
        enable_and_configure_vfs $dev 1    
    else
        skip=false
    fi
fi
```

Reply:
resolved!